src/components/Proposer/Proposer.svelte

```diff
@@ -1,17 +1,47 @@
+<script context="module" lang="ts">
+	export type PoliticianProposer = {
+		id: string;
+		firstname: string;
+		lastname: string;
+		avatar: string;
+		assembly?: {
+			id: string;
+			abbreviation: string;
+			term: number;
+			startedAt: Date;
+		};
+		partyName?: string;
+	};
+
+	export type AssemblyProposer = {
+		id: string;
+		isCabinet: boolean;
+		abbreviation: string;
+		term: number;
+		startedAt: Date;
+	};
+
+	export type PeopleProposer = {
+		ledBy: string;
+		signatoryCount: number;
+	};
+</script>
+
 <script lang="ts">
 	import GeneralIcon from '$components/icons/GeneralIcon.svelte';
 	import PeopleIcon from '$components/icons/PeopleIcon.svelte';
 	import PoliticianIcon from '$components/icons/PoliticianIcon.svelte';
-	import { AssemblyName } from '$models/assembly';
-	import type { Bill } from '$models/bill';
 	import dayjs from 'dayjs';
 	import 'dayjs/locale/th';
 	import buddhistEra from 'dayjs/plugin/buddhistEra';
 
 	dayjs.extend(buddhistEra);
 	dayjs.locale('th');
 
-	export let bill: Bill;
+	export let politician: PoliticianProposer | undefined = undefined;
+	export let assembly: AssemblyProposer | undefined = undefined;
+	export let people: PeopleProposer | undefined = undefined;
```

Comment:
I'm not so sure about this. The politician, assembly, and people can be all given at the same time, but only one will be shown according to this component logic which is not visible outside the component. It can be all undefined too.

Another solution is to combine 3 props into one with the union type and make it required. Maybe something like
```svelte
export let proposer = PoliticianProposer | AssemblyProposer | PeopleProposer;
```
That we check the type when render each case.

However, I don't have much string opinion here. What do you think?

Reply:
Hmmm... I did try that and reverted it back. I was debating with myself should I introduce a property to check type on these types or not. Let me try that and see if it makes senses. 🙌

Reply:
You can also use typescript 'in' to check if some property exists and narrow down the type without creating another flag property.

Reply:
Done. But the logic is still quite messy. I decided to move the logic under `libs/model-component-adapters` since there will be more components using this same logic. We can remove it after the whole refactoring is completed.

For the `undefined` value of the `proposer` prop, it's my own fault to design the `Bill.proposedBy*` as the current form. 😭 The component will need to accommodate `undefined` value until we get rid of them.